### ISSUE TITLE
Pluggable channel cache

### ISSUE THREAD

**Issue by jchristgit** [Kraigie/nostrum#355]
Implement pluggable channel cache, similar to the existing pluggable guild cache.

**Comment by gbottari** [comment]
Hi! We (@panoramix360 and I) are trying to develop our Elixir skills and came across this project. Maybe this issue could be a good place to start, if that's ok.

What we need to change is `Nostrum.Cache.ChannelCache` to have similar extensibility as `Nostrum.Cache.GuildCache`, right? We welcome any tips and guidelines on this.

**Comment by jchristgit** [comment]
Hi @gbottari and @panoramix360! Yes, we would welcome contributions here.

As far as guidelines go, the rough outline is to do the following:
- Find which functions a cache is required to implement (external functions)
- Create a behaviour with the functions selected above
- Move the current implementation into a "submodule", e.g. `Nostrum.Cache.ChannelCache.ETS`
- Add dispatch logic from the behaviour module to the configured cache module

.. and of course, make sure to include documentation for it 🙂. I'll assign you to the issue, let me know if you need help!

**Comment by panoramix360** [comment]
Hi @jchristgit!

I'll discuss with @gbottari so we can split the tasks and try to work together on this.

We'll let you know if we need any help :D

Thank you!

**Comment by jchristgit** [comment]
Thanks @panoramix360! Looking forward to hearing from you. 🙂

**Comment by panoramix360** [comment]
Hi @jchristgit!

We (me and @gbottari) already finished the rough implementation of what you asked for this issue :tada:

We noticed that there isn't any tests for ChannelCache so we are creating these but we had some problems with an ETS table that is needed for ChannelCache.get/1 that is only created on the Application.setup_ets_tables/0.

We didn't find any examples on using Application on a test so just for now we did the setup like this.

```
setup do
  Application.setup_ets_tables()
  [pid: start_supervised!(@cache)]
end
```
The table is `:channel_guild_map`.

If you have any suggestions, feel free to let us know :)

We hope to open the PR until friday!

Thanks
